source/common/event/file_event_impl.cc

```diff
@@ -24,6 +24,9 @@ FileEventImpl::FileEventImpl(DispatcherImpl& dispatcher, os_fd_t fd, FileReadyCb
               ? Runtime::runtimeFeatureEnabled(
                     "envoy.reloadable_features.activate_fds_next_event_loop")
               : true) {
+  // Treat the lack of a valid fd (which in practice only happens if we run out of FDs) as an OOM
+  // condition and just crash.
```

Comment:
It can also happen in the case of programmer error.

Reply:
Is it okay if I just remove the comment?